tests/test_registry_management.py

```diff
@@ -184,6 +185,42 @@ def test_auto_updates(make_test_project_dir):
         assert project_mgr.get_current_version(project_id) == VersionInfo.parse("1.2.2")
 
 
+def test_invalid_dimension_mapping(make_test_project_dir):
+    with TemporaryDirectory() as tmpdir:
+        path = create_local_test_registry(Path(tmpdir))
+        user = getpass.getuser()
+        log_message = "Initial registration"
+        manager = RegistryManager.load(path, offline_mode=True)
+
+        dim_mgr = manager.dimension_manager
+        dim_mgr.register(make_test_project_dir / "dimensions.toml", user, log_message)
+        dim_mapping_mgr = manager.dimension_mapping_manager
+        dimension_mapping_file = make_test_project_dir / "dimension_mappings.toml"
+        replace_dimension_uuids_from_registry(path, [dimension_mapping_file])
+
+        record_file = (
+            make_test_project_dir
+            / "dimension_mappings"
+            / "base-to-supplemental"
+            / "lookup_county_to_state.csv"
+        )
+        orig_text = record_file.read_text()
+
+        # Invalid 'from' record
+        record_file.write_text(orig_text + "invalid county,CO\n")
+        with pytest.raises(DSGInvalidDimensionMapping):
+            dim_mapping_mgr.register(dimension_mapping_file, user, log_message)
+
+        # Invalid 'to' record
+        record_file.write_text(orig_text.replace("CO", "Colorado"))
+        with pytest.raises(DSGInvalidDimensionMapping):
+            dim_mapping_mgr.register(dimension_mapping_file, user, log_message)
+
+        # Valid
+        record_file.write_text(orig_text)
+        dim_mapping_mgr.register(dimension_mapping_file, user, log_message)
+
+
```

Comment:
I just tested a valid 'to' record that has NULL values in the test and it fails at line 130 in `dimension_mapping_registry_manager.py` because `diff` is not None. 

This is what I added to the test to check this:
```
        # Valid 'to' record with NULLs
        record_file.write_text(orig_text.replace("CO", ""))
        dim_mapping_mgr.register(dimension_mapping_file, user, log_message)
```

Reply:
@elainethale  can you think of any other things we want to check for with these mappings? Like one thing that comes to mind is: `"What if we have something in `from` mapped to two different `to` values?"`. I think we will eventually support this, but right now we have no way of properly disaggregating/splitting the 'from' across two 'to's.

Anything else come to mind?

Reply:
I checked the scenario of `"What if we have something in 'from' mapped to two different 'to' values?"` and it fails in the tests, however I get a really confusing error message that doesn't seem to fit:`dsgrid.exceptions.DSGDuplicateValueRegistered: There are 3 dimension mappings with duplicate content (data files).`

This is the test I applied:
```
        # what if we have Denver mapped to 2 different 'to' values?
        record_file.write_text(orig_text+ "08031,TX\n")
        dim_mapping_mgr.register(dimension_mapping_file, user, log_message)
        ```

Reply:
Since to support multiple "to" values we will have to provide more data (e.g., fraction to apply to each "to") I'm glad that fails for now. I agree that the current error message is not enough to go on to figure out the problem. Is it easy to at least list what file is failing? (And I don't know why that message says that 3 mappings have duplicate content.)

I'm failing to think of any other failure conditions we want to check now, but it's also the end-ish of a long day.

Reply:
@mooneyme Did you do something like this?
```
        record_file.write_text(orig_text.replace("CO", ""))
        dim_mapping_mgr.register(dimension_mapping_file, user, log_message)

        record_file.write_text(orig_text+ "08031,TX\n")
        dim_mapping_mgr.register(dimension_mapping_file, user, log_message)
```
If so, the error message mostly makes sense. You are trying to register the same dimensions multiple times. I think that you actually wanted this (one attempt at registration with your invalid data):
```
        record_file.write_text(orig_text.replace("CO", ""))
        record_file.write_text(orig_text+ "08031,TX\n")
        dim_mapping_mgr.register(dimension_mapping_file, user, log_message)
```
This passes. We aren't checking for duplicate values in the from column. That should be a simple fix.

Reply:
@mooneyme You wrote this no-op function. I may understand what you mean for multiple to_id values. The user may want to map two different dataset records to the same project record. Why would there be multiple instances of one from record? 
```
    def validate_records(self, config: DimensionMappingsConfig, warn_only=False):
        """Validate dimension mapping records.

        Check:
        - from_id and to_id column names
        - check for duplicate IDs and log a warning if they exist (sometimes we want them to exist if it is an aggregation)
        """
        pass
```

Reply:
I think now is a good time for you all to write down the behaviors you would like to see in these cases. I suspect that our current data structures are insufficient to handle all of the cases.

Reply:
@daniel-thom 

We want to support multiple "to" values to the same "from" value. Examples:
1. Many counties map to a single state
2. TEMPO has higher fidelity of subsectors that we want to collapse at the project level

We do not want to support the SAME "to" value mapped to multiple "from" values. _At least not yet._ We are not ready for this type of disaggregation behavior because would we need to support logic for how to split out the "to" value across multiple "from" values. --> It sounds like currently dsgrid is not checking for this, but we want to fail for this at this point in time.

Reply:
To confirm: duplicate to_id values (except null) are not allowed. Duplicate from_ids are allowed. Those are obviously quite prevalent in the base-to-supplemental mappings.

Will we want different logic based on whether it is a dataset base-to-project mapping vs project base-to-supplemental mapping?

Reply:
Sorry for the confusion, I updated the language to talk about Aggregations vs. Disaggregations. :)

Reply:
This is fixed.